indico/modules/search/client/js/components/SearchApp.jsx

```diff
@@ -0,0 +1,187 @@
+
+  useEffect(() => {
+    setPage(1);
+  }, [query]);
+
+  return [
+    useMemo(
+      () => ({
+        page,
+        pages: data?.pages || 1,
+        total: data?.total || 0,
+        data: data?.results || [],
+        aggregations: data?.aggregations || lastData?.aggregations || [],
+        // ensure the initial state is loading, as the data is undefined
+        loading: (!data && !error) || loading,
+      }),
+      [page, data, lastData, error, loading]
+    ),
+    setPage,
+  ];
+}
+
+function SearchTypeMenuItem({index, active, title, total, loading, onClick}) {
+  let indicator = null;
+  if (loading) {
+    indicator = <Loader active inline size="tiny" style={{zIndex: 'unset'}} />;
+  } else if (total !== -1) {
+    indicator = ` (${total})`;
+  }
+  return (
+    <Menu.Item index={index} active={active} onClick={onClick} disabled={loading || !total}>
+      {title}
+      {indicator}
+    </Menu.Item>
+  );
+}
+
+SearchTypeMenuItem.propTypes = {
+  index: PropTypes.number.isRequired,
+  title: PropTypes.string.isRequired,
+  active: PropTypes.bool,
+  total: PropTypes.number,
+  loading: PropTypes.bool,
+  onClick: PropTypes.func,
+};
+
+SearchTypeMenuItem.defaultProps = {
+  total: 0,
+  active: false,
+  loading: false,
+  onClick: undefined,
+};
+
+function useQueryParams() {
+  const [query, _setQuery] = useState(window.location.search);
+  const queryObject = useMemo(() => Object.fromEntries(new URLSearchParams(query)), [query]);
+  const history = useHistory();
+
+  const setQuery = useCallback(
+    (type, name, reset) => {
+      const params = new URLSearchParams(reset ? undefined : query);
+      if (name !== undefined) {
+        params.set(type, name);
+      } else {
+        params.delete(type);
+      }
+      const _query = params.toString();
+      history.push({
+        pathname: window.location.pathname,
+        search: `?${_query}`,
+      });
+      _setQuery(_query);
+    },
+    [history, query]
+  );
+
+  useEffect(() => {
+    return history.listen(location => _setQuery(location.search));
+  }, [history, query, _setQuery]);
+
+  return [queryObject, setQuery];
+}
+
+export default function SearchApp() {
+  const [query, setQuery] = useQueryParams();
+  const [activeMenuItem, setActiveMenuItem] = useState(undefined);
+  const {q, ...filters} = query;
+  const [categoryResults, setCategoryPage] = useSearch(searchURL({type: 'category'}), query);
+  const [eventResults, setEventPage] = useSearch(searchURL({type: 'event'}), query);
+  const [contributionResults, setContributionPage] = useSearch(
+    searchURL({type: 'contribution'}),
+    query
+  );
+  const [fileResults, setFilePage] = useSearch(searchURL({type: 'attachment'}), query);
+  const searchMap = [
+    ['Categories', categoryResults, setCategoryPage, Category],
+    ['Events', eventResults, setEventPage, Event],
+    ['Contributions', contributionResults, setContributionPage, Contribution],
+    ['Materials', fileResults, setFilePage, File],
```

Comment:
i18n should be applied here and not below (`Translate.string(_label)` doesn't really make sense as it can't be extracted)